http/handlers.go

```diff
@@ -18,6 +18,16 @@ import (
 	"github.com/BattlesnakeOfficial/exporter/render"
 )
 
+// maxGIFResolution is the maximum resolution of GIF that we want to support.
+// This is an important limit to set, because the GIF rendering takes a lot more
+// IO, CPU and memory resources for larger resolutions.
+// This resolution was chosen as a safe upper-limit after which the rendering starts
+// to get really slow and the GIF sizes start to get too big.
+const maxGIFResolution = 504 * 504
+
+// allowedPixelsPerSquare is a list of resolutions that the API will allow.
+var allowedPixelsPerSquare = []int{10, 20, 30, 40}
```

Comment:
So this is what I came up with for restricting the GIF resolutions to limited options. Size 20 is the same as what we used to have hard-coded.

I realised that custom games can have any board size, so the restrictions would need to adapt to the board. This way every board size can have a few options that fit it well.

Reply:
A 19x19 board would only have the options of  `194x194` and `384x384`.